### PR TITLE
Improved cache-invalidation for categories/tags in excerpt tab

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR uses reference-store of categories and tags to invalidate pages where a smart-content uses tags/categories in the query.

#### To Do

- [x] Tests

### CODE REVIEW

**Comment by wachterjohannes** [comment]
closed in favor of #3777

File: src/Sulu/Bundle/PreviewBundle/Resources/config/config_preview.yml [hunk 1]
```diff
@@ -3,3 +3,4 @@ framework:
         storage_id: session.storage.mock_file
     profiler:
         enabled: false
+
```
Comment (danrot) [inline src/Sulu/Bundle/PreviewBundle/Resources/config/config_preview.yml]: Superfluous newline

File: src/Sulu/Component/HttpCache/EventSubscriber/InvalidationSubscriber.php [hunk 2]
```diff
@@ -217,6 +231,33 @@ private function invalidateDocumentUrls($document, $locale)
...
+        $extensionData = $document->getExtensionsData();
+        if (!isset($extensionData['excerpt'])) {
+            return;
+        }
+
+        $excerpt = $extensionData['excerpt'];
+
+        if (isset($excerpt['tags'])) {
+            foreach ($this->tagManager->resolveTagNames($excerpt['tags']) as $tag) {
+                $this->invalidationHandler->invalidateReference('tag', $tag);
```
Comment (danrot) [inline src/Sulu/Component/HttpCache/EventSubscriber/InvalidationSubscriber.php]: Why do we need to invalidate the entire tag? Wouldn't it be enough to only invalidate pages which have the resulting pages/snippets/contacts whatever?

Comment (wachterjohannes) [reply src/Sulu/Component/HttpCache/EventSubscriber/InvalidationSubscriber.php]: we want to invalidate pages which uses the tag/category (for example as a filter for queries - smart-content in this case).

File: src/Sulu/Component/HttpCache/Handler/AggregateHandler.php [hunk 3]
```diff
@@ -60,16 +61,40 @@ public function invalidateStructure(StructureInterface $structure)
...
         }
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function invalidateReference($alias, $id)
+    {
+        foreach ($this->handlers as $handler) {
+            if (!$handler instanceof HandlerInvalidateStructureInterface) {
```
Comment (danrot) [inline src/Sulu/Component/HttpCache/Handler/AggregateHandler.php]: Shouldn't that check be `HandlerInvalidateReferenceInterface`?

**Comment by danrot** [comment]
@trickreich I think you should also do a rebase of your caching PR quite soon, so that you can ask @wachterjohannes for help, because I guess there will be some conflicts here.

**Pull request merged**